### DISCUSSION TITLE
Is there any way to clone or modify ast?

### DISCUSSION THREAD

**Discussion by LIMPIX31** [oxc-project/oxc#2690] [Q&A] [answered]
I want to get owned `Program` to be able modify it. But the `Program` has `Box`es that not owns the data, only stores `&'alloc mut T`

**Comment by Boshen** [comment, marked as answer]
The transformer mutates the AST, for example https://github.com/oxc-project/oxc/blob/86ee074678eeab786dc80cbc2c1c7d0efd763a98/crates/oxc_transformer/src/es2022/class_static_block.rs#L77-L101

Mutating a lifetimed struct with allocators in Rust is kind of painful, so ideally you want to use a builder pattern for the AST nodes https://github.com/oxc-project/oxc/blob/main/crates/oxc_ast/src/ast_builder.rs
